### ISSUE TITLE
Python 3 Compatibility

### ISSUE THREAD

**Issue by avorio** [bloomonkey/oai-harvest#10]
Hello!

I would like to use oai-harvest in a Python 3 environment. Is there any interest from somebody else in porting it? Has this been considered before?

Many thanks.

**Comment by bloomonkey** [comment]
I've not looked at porting it to Python 3.x. In fact I've not had much time, or excuse to work on oai-harvest recently :|

I'd be happy to receive any pull request that ported to 3.x while maintaining backward compatibility with 2.7.

**Comment by bloomonkey** [comment]
It looks like the [latest version of pyoai](https://pypi.python.org/pypi/pyoai) - on which oai-harvest depends - is not compatible with Python 3 so it is not possible to support it at this time. Sorry :(

**Comment by ulikoehler** [comment]
I've fixed most Python3 bugs in https://github.com/bloomonkey/oai-harvest/pull/14 , so now I can harvest CiteseerX using Python3. Maybe there are still bugs left in other parts of `oai-harvest` or `oai-reg`. Backwards compatibility is provided using `six`.

@avorio Would it be possible for you to test some of your usecases on my branch?
See my branch here: https://github.com/ulikoehler/oai-harvest/tree/py3

`pyoai` has already been fixed on GitHub and I've filed an issue there to release a new version on PyPI. In the meantime, the README lists a `pip` command that allows you installing the GitHub version of `pyoai`
